mammoth/utils/optimizers.py

```diff
@@ -254,30 +255,25 @@ def from_opts(cls, model, opts, task_queue_manager, checkpoint=None):
         )
 
     @staticmethod
-    def _maybe_restore_from_checkpoint(opts, checkpoint=None):
+    def _maybe_restore_from_checkpoint(opts, frame_checkpoint=None):
         optim_opts = opts
-        optim_state_dict = None
 
-        if opts.train_from and checkpoint is not None:
-            ckpt_opts = checkpoint['opts']
-            ckpt_state_dict = checkpoint['optim']
+        if opts.train_from and frame_checkpoint is not None:
+            ckpt_opts = frame_checkpoint['opts']
 
             if opts.reset_optim == 'none':
                 # Load everything from the checkpoint.
                 optim_opts = ckpt_opts
-                optim_state_dict = ckpt_state_dict
             elif opts.reset_optim == 'all':
                 # Build everything from scratch.
                 pass
             elif opts.reset_optim == 'states':
                 # Reset optimizer, keep options.
                 optim_opts = ckpt_opts
-                optim_state_dict = ckpt_state_dict
-                del optim_state_dict['optimizer']
             elif opts.reset_optim == 'keep_states':
                 # Reset options, keep optimizer.
-                optim_state_dict = ckpt_state_dict
-        return optim_opts, optim_state_dict
+                pass
```

Comment:
? either remove the cond branch and add a comment, or impl this pass?

Reply:
Does anyone actually need the `keep_states` setting, or was it added just for completeness and can be removed?
It used to be convenient to do it here, but now options and optimizer parameters are handled in different places, so this branch does nothing here.
`load_parameters_from_checkpoint` in `model_saver` handles the other part.